File: app/src/main/java/com/lucasurbas/masterdetail/ui/util/BaseActivity.java
package com.lucasurbas.masterdetail.ui.util;

import android.os.Bundle;

import com.lucasurbas.masterdetail.app.MasterDetailApplication;
import com.lucasurbas.masterdetail.injection.app.ApplicationComponent;

import androidx.appcompat.app.AppCompatActivity;

/**
 * Created by Lucas on 19/06/16.
 */
public abstract class BaseActivity extends AppCompatActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setupActivityComponent(MasterDetailApplication.getAppComponent(this));
    }

    protected abstract void setupActivityComponent(ApplicationComponent applicationComponent);
}

File: app/src/main/java/com/lucasurbas/masterdetail/ui/util/BaseActivity.kt
package com.lucasurbas.masterdetail.ui.util

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.lucasurbas.masterdetail.app.MasterDetailApplication
import com.lucasurbas.masterdetail.injection.app.ApplicationComponent
import org.jetbrains.annotations.NotNull


abstract class BaseActivity : AppCompatActivity() {


    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setupActivityComponent(MasterDetailApplication.getAppComponent(this))
    }

    protected abstract fun setupActivityComponent(@NotNull applicationComponent: ApplicationComponent)

}
